acceptance/tests/plan_winrm.rb

```diff
@@ -7,64 +7,109 @@
-      notice("${run_task(test::a_win, $nodes_array)}")
-      notice("${run_task(test::b_win, $nodes_array)}")
-    }
+                       "#{dir}/modules/test/plans/winrm_retry_plan.pp", <<-FILE)
+plan test::winrm_retry_plan($nodes) {
+  $node_array = $nodes.split(',')
+  $result = run_task(test::hostname, $node_array,
+        '_catch_errors' =>true)
+  $retry = run_task(test::hostname, $result.error_set.names)
+  return({ 'result' => $result, 'retry' => $retry })
+}
     FILE
   end
 
-  step "execute `bolt plan run` via WinRM" do
-    user = ENV['WINRM_USER']
-    password = ENV['WINRM_PASSWORD']
-    nodes_csv = winrm_nodes.map { |host| "winrm://#{host.hostname}" }.join(',')
+  step "execute `bolt plan run` via WinRM with json output" do
     bolt_command = "bolt plan run test::my_win_plan nodes=#{nodes_csv}"
-
     flags = {
       '--modulepath'  => "#{dir}/modules",
       '-u'            => user,
       '-p'            => password,
+      '--format'      => 'json',
       '--no-ssl'      => nil
     }
 
     result = bolt_command_on(bolt, bolt_command, flags)
-    message = "The plan was expected to notify but did not"
-    assert_match(/Notice:/, result.stdout, message)
-    winrm_nodes.each do |node|
-      message = "The plan was expected to mention the host #{node.hostname} with _output"
-      assert_match(/#{node.hostname}","status":"success"/, result.stdout, message)
+    assert_equal(0, result.exit_code,
+                 "Bolt did not exit with exit code 0")
+
+    begin
+      json = JSON.parse(result.stdout)
+    rescue JSON.ParserError
+      assert_equal("Output should be JSON", result.string,
+                   "Output should be JSON")
     end
 
-    winrm_nodes.each do |node|
-      command = "type #{testdir}/C100554_plan_artifact.txt"
-      on(node, powershell(command), accept_all_exit_codes: true) do |res|
-        type_msg = "The powershell command 'type' was not successful"
-        assert_equal(res.exit_code, 0, type_msg)
-
-        msg = 'The expected contents of the plan artifact were not observed'
-        assert_match(/Line one from task a/, res.stdout, msg)
-        assert_match(/Line two from task b/, res.stdout, msg)
-      end
+    # Verify that the first node failed on the first run
+    failed_node = json['result'].select { |n| n['status'] == 'failure' }
+    assert(!failed_node.empty?, "No nodes failed on the first task run")
+    assert(failed_node.length < 2, "More than 1 node failed the first task run")
+    assert_equal("winrm://#{winrm_nodes[0].hostname}", failed_node[0]['node'],
+                 "The hostname #{winrm_nodes[0].hostname} is not correct")
+
+    # Verify that the second node succeeded on the first run
+    if winrm_nodes.length > 1
+      success_nodes = json['result'].select { |n| n['status'] == 'success' }
+      assert_equal(winrm_nodes.length - 1, success_nodes.length,
+                   "Some nodes did not succeed in the first task run")
```

Comment:
This message seems a bit confusing. perhaps we should look for specific nodes and generate an error that displays the unexpected result.

Reply:
M'k, I modified it to look through the remaining nodes and individually verify that they succeeded, then error with the specific node that didn't succeed if it exists.